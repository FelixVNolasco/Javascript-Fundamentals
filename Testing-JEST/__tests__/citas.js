import Citas from '../js/classes/Citas';

describe("Probando la clase de citas", ()=> {

    const citas = new Citas();
    const id  = Date.now();    
    test("Probando funcionalidad de agregar una nueva cita", ()=> {
        const citaObj = {
            id,
            mascota: 'Kira',
            propietario: 'Felix Vega',
            telefono: '5578136020',
            fecha: '15 de mayo',
            hora:'13:00',
            sintomas: 'baño'
        };        
        citas.agregarCita(citaObj);
        // expect(citas).toMatchSnapshot();

    });
    test('Probando funcionalidad de actualizar', () => {
        const citaActualizada = {
            id,
            mascota: 'Chatita',
            propietario: 'Felix Vega',
            telefono: '5578136020',
            fecha: '15 de mayo',
            hora:'13:00',
            sintomas: 'baño'
        };  
        citas.editarCita(citaActualizada)
        // expect(citas).toMatchSnapshot();        
    });
    test('Probando funcionalidad de eliminar', () => {
        citas.eliminarCita(id);  
        // expect(citas).toMatchSnapshot();
    })
    
    
})