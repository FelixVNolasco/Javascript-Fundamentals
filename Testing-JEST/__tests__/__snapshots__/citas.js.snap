// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando la clase de citas Probando funcionalidad de actualizar 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "15 de mayo",
      "hora": "13:00",
      "id": 1619726710258,
      "mascota": "Chatita",
      "propietario": "Felix Vega",
      "sintomas": "baño",
      "telefono": "5578136020",
    },
  ],
}
`;

exports[`Probando la clase de citas Probando funcionalidad de agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "15 de mayo",
      "hora": "13:00",
      "id": 1619726710258,
      "mascota": "Kira",
      "propietario": "Felix Vega",
      "sintomas": "baño",
      "telefono": "5578136020",
    },
  ],
}
`;

exports[`Probando la clase de citas Probando funcionalidad de eliminar 1`] = `
Citas {
  "citas": Array [],
}
`;
